src/main/java/edu/cornell/kfs/pdp/businessobject/lookup/CuPayeeACHAccountSearchService.java

```diff
@@ -0,0 +1,195 @@
+            final String sortField) {
+        
+        final List<Person> people = findPeopleByPrincipalName(principalName);
+        if (people.isEmpty()) {
+            LOG.debug("searchByPrincipalName, No people found for principal name: {}", principalName);
+            return new ArrayList<>();
+        }
+
+        // Extract entity and employee IDs
+        final Pair<List<String>, List<String>> ids = extractPersonIds(people);
+        final List<String> entityIds = ids.getLeft();
+        final List<String> employeeIds = ids.getRight();
+
+        // Build predicate for payee ID search
+        final Predicate principalNamePredicate = buildPrincipalNamePredicate(entityIds, employeeIds);
+
+        // Execute the search
+        final List<PayeeACHAccount> results = executePayeeSearch(searchProps, skip, principalNamePredicate);
+
+        // Sort results
+        return sortResults(results, sortField);
+    }
+    
+    private List<Person> findPeopleByPrincipalName(final String principalName) {
+        final Map<String, String> searchCriteria = Collections.singletonMap(
+            KIMPropertyConstants.Principal.PRINCIPAL_NAME, principalName);
+        return personService.findPeople(searchCriteria);
+    }
+    
+    private Pair<List<String>, List<String>> extractPersonIds(final List<Person> people) {
+        final List<String> entityIds = new ArrayList<>();
+        final List<String> employeeIds = new ArrayList<>();
+        
+        for (final Person person : people) {
+            if (StringUtils.isNotBlank(person.getEntityId())) {
+                entityIds.add(person.getEntityId());
+            }
+            if (StringUtils.isNotBlank(person.getEmployeeId())) {
+                employeeIds.add(person.getEmployeeId());
+            }
+        }
+        
+        return Pair.of(entityIds, employeeIds);
+    }
+    
+    private Predicate buildPrincipalNamePredicate(final List<String> entityIds, final List<String> employeeIds) {
+        // Build the sub-predicate to limit by the entity or employee IDs for the given
+        // principal names.
+        final Predicate principalNameEquivalentPredicate;
+        if (employeeIds.isEmpty()) {
+            principalNameEquivalentPredicate = PredicateFactory.and(
+                    PredicateFactory.equal(PdpPropertyConstants.PAYEE_IDENTIFIER_TYPE_CODE, PayeeIdTypeCodes.ENTITY),
+                    PredicateFactory.in(PdpPropertyConstants.PAYEE_ID_NUMBER,
+                            entityIds.toArray(new String[entityIds.size()])));
+        } else {
+            principalNameEquivalentPredicate = PredicateFactory.or(PredicateFactory.and(
+                    PredicateFactory.equal(PdpPropertyConstants.PAYEE_IDENTIFIER_TYPE_CODE, PayeeIdTypeCodes.ENTITY),
+                    PredicateFactory.in(PdpPropertyConstants.PAYEE_ID_NUMBER,
+                            entityIds.toArray(new String[entityIds.size()]))),
+                    PredicateFactory.and(
```

Comment:
The current formatting of this predicate-building section is a little hard to read. I would suggest moving the first opening `PrediateFactory.and(` call onto its own line and indenting its sub-predicates further, like in the equivalent code from the old lookupable helper service.